hooks/open-telemetry/pkg/otel.go

```diff
@@ -2,165 +2,52 @@ package otel
 
 import (
 	"context"
-	"encoding/json"
-	"errors"
-	"fmt"
-	"sync"
 
-	of "github.com/open-feature/go-sdk/pkg/openfeature"
+	"github.com/open-feature/go-sdk/pkg/openfeature"
 	"go.opentelemetry.io/otel/attribute"
-	"go.opentelemetry.io/otel/codes"
 	"go.opentelemetry.io/otel/trace"
 )
 
 const (
-	// AttributeFlagKey trace attribute name for the evaluated flag key
-	AttributeFlagKey = "feature_flag.flag_key"
-	// AttributeProviderName trace attribute name for the provider used for evaluation
-	AttributeProviderName = "feature_flag.provider_name"
-	// AttributeEvaluatedVariant trace attribute name for the variant returned from the flag evaluation
-	AttributeEvaluatedVariant = "feature_flag.evaluated_variant"
-	// AttributeEvaluatedValue trace attribute name for the value returned from the flag evaluation
-	AttributeEvaluatedValue = "feature_flag.evaluated_value"
-	// AttributeTraceName the name of the trace exported by the hook
-	AttributeTraceName = "github.com/open-feature/go-sdk-contrib/hooks/open-telemetry/pkg"
+	EventName                 = "feature_flag"
+	EventPropertyFlagKey      = "feature_flag.key"
+	EventPropertyProviderName = "feature_flag.provider_name"
+	EventPropertyVariant      = "feature_flag.variant"
 )
 
 type Hook struct {
-	spans        map[string]*mutexWrapper
-	ctx          context.Context
-	wg           *sync.WaitGroup
-	tracerClient tracerClientInterface
-}
-
-// mutex wrapper is used to prevent colliding keys from overwriting each other / closing a partially completed span
-type mutexWrapper struct {
-	mu *sync.Mutex
-	ss *storedSpan // ss has been set as a pointer so it can be set to nil during cleanup
-}
-type storedSpan struct {
-	cancel func()
-	span   trace.Span
+	ctx context.Context
+	openfeature.UnimplementedHook
 }
 
 // NewHook return a reference to a new instance of the OpenTelemetry Hook
-func NewHook() *Hook {
+func NewHook(ctx context.Context) *Hook {
```

Comment:
Should we somehow warn of an invalid state if this constructor is not used and the context is unset?

Reply:
Unexpected the hook struct so the use of `NewHook` is forced, past this point I think the log is unnecessary, as you would have to pass a nil context into the constructor as you can no longer bypass this by creating the hook directly with `hook := otelhook.Hook{}`